File: doc-examples/nitrite-example-java/src/main/java/example/CatalogItem.java
package example;

import io.micronaut.data.annotation.GeneratedValue;
import io.micronaut.data.annotation.Id;
import io.micronaut.data.annotation.Index;
import io.micronaut.data.annotation.MappedEntity;

// tag::unique-index[]
@MappedEntity
public class CatalogItem {
    @Id
    @GeneratedValue
    private String id;

    @Index(columns = "sku", unique = true)
    private String sku;

    private String name;

    public CatalogItem() {
    }

    public CatalogItem(String sku, String name) {
        this.sku = sku;
        this.name = name;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getSku() {
        return sku;
    }

    public void setSku(String sku) {
        this.sku = sku;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }
}
// end::unique-index[]

File: doc-examples/nitrite-example-kotlin/src/main/kotlin/example/CatalogItem.kt
package example

import io.micronaut.data.annotation.GeneratedValue
import io.micronaut.data.annotation.Id
import io.micronaut.data.annotation.Index
import io.micronaut.data.annotation.MappedEntity

// tag::unique-index[]
@MappedEntity
class CatalogItem {
    @Id
    @GeneratedValue
    var id: String? = null

    @Index(columns = ["sku"], unique = true)
    var sku: String? = null

    var name: String? = null

    constructor()

    constructor(sku: String, name: String) {
        this.sku = sku
        this.name = name
    }
}
// end::unique-index[]
